core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardServiceTest.java

```diff
@@ -264,43 +268,34 @@ public void testRemoveDSWError() throws Exception {
   }
 
   @Test
-  public void testGetDSWDatasourceIds() throws Exception {
-    final String TEST_DOMAIN_ID = "mock";
-
-    LogicalModelSummary mockLogicalModelSummary  = mock( LogicalModelSummary.class );
-    doReturn(TEST_DOMAIN_ID).when(mockLogicalModelSummary).getDomainId();
-    List<LogicalModelSummary> mockLogicalModelSummaryList = new ArrayList<LogicalModelSummary>();
-    mockLogicalModelSummaryList.add( mockLogicalModelSummary );
-
-    List<String> datasourceList = new ArrayList<String>();
-    datasourceList.add( mockLogicalModelSummary.getDomainId() );
-
-    doReturn( mockLogicalModelSummaryList ).when( dataSourceWizardService.dswService ).getLogicalModels( null );
-
-    Map<String,InputStream> mockDomainFilesData = new HashMap<String,InputStream>();
-
-    mockDomainFilesData.put(TEST_DOMAIN_ID, IOUtils.toInputStream("AGILE_BI_GENERATED_SCHEMA"));
-    doReturn(mockDomainFilesData).when((IPentahoMetadataDomainRepositoryExporter)dataSourceWizardService.metadataDomainRepository).getDomainFilesData(anyString());
-
-    List<String> datasourceIds = dataSourceWizardService.getDSWDatasourceIds();
-    verify( dataSourceWizardService, times( 1 ) ).getDSWDatasourceIds();
-    assert( datasourceIds.contains(TEST_DOMAIN_ID));
+  public void testGetDSWDatasourceIds() {
+    List<String> mockMetadataIdsList = new ArrayList<String>();
+    Set<String> mockSet = new HashSet<String>();
+    mockSet.add( "domainId1" );
+    mockMetadataIdsList.add( "domainId1" );
+    doReturn( 1 ).when( dataSourceWizardService ).noOfThreads();
+    doReturn( "domainId1" ).when( dataSourceWizardService ).isDataSourceWizard( "domainId1" );
```

Comment:
I don't think this is a valid unit test, when you mock out functions of the class under test.  In this case,`DataSourceWizardService#isDataSourceWizard(string)`

You are essentials just testing that your mocks interact with each other as expected in the test case. The underlying source code could have change, but you'll never know because the mocked tests pass.

Changes I would make:

1. you shouldn't mock `dataSourceWizardService#noOfThreads`, it will get a non null number regardless.
2. don't mock `dataSourceWizardService#isDataSourceWizard`, instead mock `metadataDomainRepository.getDomain( id );` line and create an instance of Domain that will pass